utils/utils.py

```diff
@@ -81,7 +81,7 @@ def sanity_check(args: Union[argparse.Namespace, DictConfig]):
         print('For instance, in one terminal: `device="cuda:0"` and `device="cuda:1"` in the second, etc.')
         print(f'Your device specification (device_ids={args.device_ids}) is converted to `device="cuda:0"`.')
         args.device = 'cuda:0'
-    if 'cuda' in args.device and not torch.cuda.is_available():
+    if 'cuda' in args.device and not torch.cuda.is_available() and args.slurm.submit == False:
```

Comment:
what happens if we use slurm (`submit` is `True`) but `cuda` is not available? i am not sure why the new condition was added

Reply:
Because of the timing of when you check the args -- specifically before you create the extractor instance.
In this case, you check whether there is cuda (which would likely be false on the login node so it forces cpu), and then you submit the  cpu-instances to the cluster nodes...